book/012-vis-adv.Rmd

```diff
@@ -1 +1,155 @@
 # {ggplot2} advanced {#vis-adv}
+
+## Facetting
+
+<details><summary>*Click here to show setup code.*</summary>
+```{r setup-vis-adv}
+library(tidyverse)
+library(ggsci)
+library(ggpubr)
+library(patchwork)
+```
+</details>
+
+<br>
+
+"Facetting" (or trellis plots, lattice plots) denotes an idea of dividing a graphic into sub-graphics based on the (categorical) values of one or more variables of a dataset.
+
+```{r, eval=FALSE}
+facet_grid(facets = . ~ ., scales = "fixed", ...)
+facet_wrap(facets = <variable> ~ ., scales = "fixed", ...)
+```
+
+**facet**: Formula with splitting variable.
+
+**scales**: Scale of the axes over the sub-graphics.
+
+The position of `<variable>` denotes on which axis the facets will be shown:
+
+- `<variable> ~ .` `r icon::fa_arrow_right()` y-axis
+
+- `. ~ <variable> ``r icon::fa_arrow_right()` x-axis
+
+![](img/facets.png)
+
+### `facet_wrap()`
+
+```{r, fig.width=8, fig.height=4}
+facet_x = ggplot(mpg, aes(displ, hwy)) +
+  geom_point(aes(colour = fl)) +
+  scale_color_nejm() +
+  facet_grid(. ~ year) +
+  theme_pubr()
+
+facet_y = ggplot(mpg, aes(displ, hwy)) +
+  geom_point(aes(colour = fl)) +
+  scale_color_nejm() +
+  facet_grid(year ~ .) +
+  theme_pubr()
+```
+
+```{r}
+facet_x + facet_y
+```
+
+Facets can also be shown on both axes using different (categorical) variables
+
+```{r, fig.width=8, fig.height=4}
+ggplot(mpg, aes(displ, hwy)) +
+  geom_point(aes(colour = class)) +
+  facet_grid(year ~ cyl) +
+  theme_pubr()
+```
+
+Rather than visualizing a 2D-facet plot on x and y, there is also the option to combine both in one axis.
+(For this to work, the variables need to be of class `factor`).
+
+```{r}
+mpg$manufacturer = as.factor(mpg$manufacturer)
```

Comment:
I'd like to avoid creating tweaked copies of datasets. Can we do

```r
mpg %>%
  mutate(...) %>%
  ggplot(...) ...
```

?

Reply:
sure